src/XRP/default-xrp-client.ts

```diff
@@ -156,11 +156,11 @@ export default class DefaultXrpClient implements XrpClientDecorator {
    * for additional details to be specified for use with supplementary features of the XRP ledger.
    *
    * @param sendXrpDetails - a wrapper object containing details for constructing a transaction.
-   * @returns A promise which resolves to a string representing the hash of the submitted transaction.
+   * @returns A promise which resolves to a TransactionResult representing the pending outcome of the submitted transaction.
    */
-  public async sendWithDetails(
+  public async sendXrpWithDetails(
     sendXrpDetails: SendXrpDetails,
-  ): Promise<string> {
+  ): Promise<TransactionResult> {
```

Comment:
So here in `DefaultXrpClient` I had a decision to make... Currently, `DefaultXrpClient` implements `XrpClientDecorator`, which means that the return value of this method has to be the same as the return value in `ReliableSubmissionXrpClient` AKA a `TransactionResult`.  However, at this point in the process, all we actually need is the transaction hash... the other result values like the status and validated/unvalidated don't matter yet because it's for sure pending...  So what I COULD do is restore this to the original string return and then have `DefaultXrpClient` not implement `XrpClientDecorator` (because it's not actually a decorator... it doesn't decorate anything).  Or I can just leave it like this, I'm not sure it really matters.

Reply:
i think it's fine to use `TransacitonResult`. i like the more consistent interface...when i tried to use send it felt strange to me that it returned a string instead of `TransactionResult`